### ISSUE TITLE
Identify Node Type and accessing AST Node

### ISSUE THREAD

**Issue by parvez2014** [GumTreeDiff/gumtree#111]
Thanks for the GumTree again!
I am using the following code to determine what changes have been made and collect the node.
if(action instanceof Addition) {
	Addition addition = (Addition)action;
        ITree node = addition.getNode() 
}
Could you pls. answer the following questions or point me a direction:
a) The getType method of the node gives me an integer number. How can I identify what the number mean (like whether this is a method call, SimpleName or a variableDeclarationExpression)
b) Is it possible to get access to the underlying AST node?
c) What is the difference between addition and insert?

**Comment by 11510746** [comment]
> it's not possible to access the underlying node because we do not create a link at parsing with the parser node.

Will this feature available in the future?

**Comment by monperrus** [comment]
In Java, this feature is already available in https://github.com/SpoonLabs/gumtree-spoon-ast-diff/

**Comment by jrfaller** [comment]
Yes you can look into gumtree-spoon. However this feature is very hard to generalize, it is not possible at all for all external parsers, and hard to achieve with some Java parsers.

**Comment by monperrus** [comment]
> this feature is very hard to generalize

Fully agree, this feature is language specific
